lib/lint.ml

```diff
@@ -13,6 +13,33 @@ module Check = struct
   let marshal () = Yojson.Safe.to_string `Null
   let unmarshal _ = ()
 
+  let parse_errors_from_log job =
+    let job_id = Current.Job.id job in
+    let path = Current.Job.log_path job_id in
+    match path with
+    | Ok path ->
+      In_channel.with_open_text (Fpath.to_string path) (fun ic ->
+          let rec aux acc =
+            match In_channel.input_line ic with
+            (* We choose the relevant output lines by filtering lines starting
+               with Error/Warning. The other log lines in the file would start
+               with a timestamp (logged info), or not start with the
+               Error/Warning prefix (other text output from the CLI tool). *)
+            | Some line ->
+              if (String.starts_with ~prefix:"Error" line ||
+                 String.starts_with ~prefix:"Warning" line)
+              then
+                aux (line :: acc)
+              else
+                aux acc
+            | None -> acc
+          in
+          let errors = aux [] in
+          match errors with
+          | [] -> Error (`Msg "Could not find any Error/Warning lines in CLI output.")
+          | _ -> Ok (String.concat "\n" errors) )
+    | Error _ -> Error (`Msg (Fmt.str "Could not find the log file for job <%s>" job_id))
```

Comment:
Should we try to preserve the err data given by `log_path` in case that is useful for debugging? E.g.,

```suggestion
    | Error (`Msg m) -> Error (`Msg (Fmt.str "Could not find the log file for job <%s>: %s" job_id m))
```